src/aws_s3_client_module.js

```diff
@@ -19,46 +20,29 @@ const {S3, Endpoint} = pkg;
  */
 
 async function uploadFileToBucket (accessKeyId, secretAccessKey, region, file, bucket, url) {
-    return new Promise((resolve, reject)=>{
-        const fileName = path.basename(file);
-
-        // Initialized the AWS S3 Client
+    const uploadStream = ({ Bucket, Key }) => {
         const s3Client = new S3({
             accessKeyId: accessKeyId,
             secretAccessKey: secretAccessKey,
             endpoint: new Endpoint(
                 joinURL('https://', region + url)
             )
         });
-
-        // Read the file contents and upload it to object storage
-        readFile(path.resolve(file), {}, async (error, data) => {
-            if (error) {
-                console.error('There was a error reading your file\n' + JSON.stringify(error));
-                reject(error);
-                return;
-            }
-
-            console.log('Uploading...');
-
-            try {
-                await s3Client
-                    .putObject({
-                        Bucket: bucket,
-                        Key: fileName,
-                        Body: data
-                    })
-                    .promise();
-
-                console.log(`Uploaded ${file}`);
-                resolve(file);
-            } catch (e) {
-                const errorMsg = 'Error in uploading\n' + JSON.stringify(e);
-                console.error(errorMsg);
-                reject(e);
-            }
-        });
-    });
+        const pass = new stream.PassThrough();
+        return {
+            writeStream: pass,
+            promise: s3Client.upload({ Bucket, Key, Body: pass }).promise()
+        };
+    };
+    const filePath = path.basename(file);
+    const { writeStream, promise } = uploadStream({Bucket: bucket, Key: filePath});
+    fsExtra.createReadStream(filePath).pipe(writeStream);
+    let response = {
+        status: true,
+        errorMessage: ''
+    };
+    await promise.catch((reason)=> { response.status = false; response.errorMessage(reason.toString()); });
```

Comment:
Good to split into multiple lines to step with debugger to debug error messages. {} to 
{
}.

Reply:
Will this line succeed for both success and exception case?

Reply:
yup tested it out using the code on driver.js, intentionally manipulated the params to throw an exception, Got the following message

`1405086 (test) src $ node driver.js
node:events:368
      throw er; // Unhandled 'error' event
      ^

Error: ENOENT: no such file or directory, open 'C:\Users\Admin\Documents\Work\li
node-object-storage-lib\src\YOUR_FILE_NAME'
Emitted 'error' event on ReadStream instance at:
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'open',
  path: 'C:\\Users\\Admin\\Documents\\Work\\linode-object-storage-lib\\src\\YOUR
_FILE_NAME'`